machinery-helper/machinery_helper.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2015 SUSE LLC
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of version 3 of the GNU General Public License as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE about this file by physical or electronic mail,
+// you may find current contact information at www.suse.com
+
+package main
+
+import (
+  "os/exec"
+  "bytes"
+  "log"
+  "strings"
+  "io/ioutil"
+  "encoding/json"
+  "os"
+  "sort"
+  "fmt"
+  "path/filepath"
+  "unicode/utf8"
+  "flag"
+)
+
+var Ignore_list map[string]bool
+var RemoteMount_list map[string]bool
+
+func getRpms() []string {
+  cmd := exec.Command("rpm", "-qlav")
+  var out bytes.Buffer
+  cmd.Stdout = &out
+  err := cmd.Run()
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  f := func(c rune) bool {
+    return c == '\n'
+  }
+  packages := strings.FieldsFunc(out.String(), f)
+  return packages
+}
+
+func parseRpmLine(line string) (file_type string, file_name string, link_target string) {
```

Comment:
don't use underscores in Go names; func result file_type should be fileType<br>don't use underscores in Go names; func result file_name should be fileName<br>don't use underscores in Go names; func result link_target should be linkTarget